### DISCUSSION TITLE
How to pin a third party application using WinUI3

### DISCUSSION THREAD

**Discussion by wagnerbit** [microsoft/microsoft-ui-xaml#9909] [General]
Hello, guys!
I want to create an application that works as a hub to access and pin third-party applications installed on the operating system to the taskbar.

Currently, I can meet my requirements using UWP. 
await TaskbarManager.GetDefault().RequestPinAppListEntryAsync((AppListEntry)appToPin);

I followed the steps in the tutorial below, including requesting the token to create the LafData file, as the API is part of a series with a Limited Access Feature (LAF). 
However, I continue to have permission problems pinning applications to the taskbar. If anyone can give me an idea of ​​how I could use this functionality in a WinUI3 application?
https://learn.microsoft.com/en-us/windows/apps/design/shell/pin-to-taskbar

**Comment by Gaoyifei1011** [comment]
> Hello, guys! I want to create an application that works as a hub to access and pin third-party applications installed on the operating system to the taskbar.
> 
> Currently, I can meet my requirements using UWP. await TaskbarManager.GetDefault().RequestPinAppListEntryAsync((AppListEntry)appToPin);
> 
> I followed the steps in the tutorial below, including requesting the token to create the LafData file, as the API is part of a series with a Limited Access Feature (LAF). However, I continue to have permission problems pinning applications to the taskbar. If anyone can give me an idea of ​​how I could use this functionality in a WinUI3 application? https://learn.microsoft.com/en-us/windows/apps/design/shell/pin-to-taskbar

这是我在我的应用中使用 taskbarmanager api固定应用到任务栏的一个示例，这个api需要访问受限功能。注意，如果在桌面应用使用这个api，通过访问受限功能只能固定当前应用

This is an example of my application using the taskbarmanager api to fix to the taskbar, which requires access restriction functionality. Note that if you use this api in a desktop application, you can only fix the current application with restricted access

请注意，这是一个受限访问功能。如果需要提交应用到 Microsoft store，您可能需要解释为什么需要访问这个受限功能😂

Please note that this is a restricted access feature. If you need to submit your app to the Microsoft store, you might need to explain why you need to access this limited feature 😂

https://github.com/Gaoyifei1011/GetStoreApp/blob/main/GetStoreApp/Views/Pages/AboutPage.xaml.cs#L202

![Screenshot_2024-08-20-00-49-05-836_mark via](https://github.com/user-attachments/assets/ccb37e80-084c-4b2c-9626-56f58ad078df)

**Reply by wagnerbit** [reply]
Thanks @Gaoyifei1011 Thank you very much, this will certainly help me.

**Comment by castorix** [comment]
A way is with **IPinnedList3** interface, like in [Chromium](https://github.com/chromium/chromium/blob/ecf049fba95834f05e06cfbc6388dc435fe5f0e8/chrome/installer/util/taskbar_util.cc#L89)
I tested with a random .exe and it works on my Windows 10 22H2 OS (there is a Popup asking if I want to pin the app)
Without Popup, it can be done with _CreateThread_ to inject code into Explorer simulating the Context menu "Pin to taskbar" (_IContextMenu.InvokeCommand_), but it is more complicated...

**Reply by wagnerbit** [reply]
Thanks for the tip, @castorix ! I'm sure your tip will help me.
